src/QAST/Node.nqp

```diff
@@ -107,11 +107,13 @@ class QAST::Node {
         %!annotations := nqp::null();
     }
 
-    my %uniques;
+    my %global-uniques;
     method unique($prefix) {
+        my %uniques := nqp::clone(%global-uniques);
         my $id := nqp::existskey(%uniques, $prefix) ??
```

Comment:
Incidental to this PR, but this is very old code and maybe it predates hash auto-vivication? Because this line could be just `my $id := ++%uniques{$prefix}`

Reply:
I thought we didn't have auto-vivification in NQP ?